src/domain/products/components/variant-form/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React from "react"
+import { Controller, useFieldArray, UseFormReturn } from "react-hook-form"
+import Switch from "../../../../components/atoms/switch"
+import IconTooltip from "../../../../components/molecules/icon-tooltip"
+import InputField from "../../../../components/molecules/input"
+import Accordion from "../../../../components/organisms/accordion"
+import FormValidator from "../../../../utils/form-validator"
+import { nestedForm } from "../../../../utils/nested-form"
+import CustomsForm, { CustomsPayload } from "../customs-form"
+import DimensionsForm, { DimensionsPayload } from "../dimensions-form"
+import PricesForm, { PricesPayload } from "../prices-form"
+
+export type VariantFormType = {
+  prices: PricesPayload
+  manage_inventory: boolean
+  allow_backorder: boolean
+  title: string | null
+  sku: string | null
+  ean: string | null
+  upc: string | null
+  barcode: string | null
+  options: {
+    id: string
+    value: string
+    title: string
+  }[]
+  inventory_quantity: number | null
+  material: string | null
+  dimensions: DimensionsPayload
+  customs: CustomsPayload
+}
+
+type Props = {
+  form: UseFormReturn<VariantFormType, any>
+}
+
+/**
+ * Re-usable Product Variant form used to add and edit product variants.
+ * @example
+ * const MyForm = () => {
+ *   const form = useForm<VariantFormType>()
+ *   const { handleSubmit } = form
+ *
+ *   const onSubmit = handleSubmit((data) => {
+ *     // do something with data
+ *   })
+ *
+ *   return (
+ *     <form onSubmit={onSubmit}>
+ *       <VariantForm form={form} />
+ *       <Button type="submit">Submit</Button>
+ *     </form>
+ *   )
+ * }
+ */
+const VariantForm = ({ form }: Props) => {
+  const { fields } = useFieldArray({
+    control: form.control,
+    name: "options",
+  })
+
+  return (
+    <Accordion type="multiple" defaultValue={["general"]}>
```

Comment:
question: Upon expanding an accordion, do we want all others to be closed? This is not the current behaviour but I believe that is the conventional definition of an accordion - please correct me if I am wrong.

We could have an option for the accordion component defining whether or not multiple can be open at the same time.

Reply:
We have that prop on the component, I set it to allow multiple open right now, but we can definetly change it. I personally think it would be a bit annoying, but fine with it if you think its the way to go :)

Reply:
Ahh got it! Sorry, missed that. Think it's fine this way 👍